### ISSUE TITLE
Errors caused by refute statement inside of loop with goto

### ISSUE THREAD

**Issue by bruggerl** [viperproject/silicon#773]
When verifying the following method, Silicon throws an exception (leading to a "Verification aborted exceptionally" error) _and_ also falsely reports that the `refute` statement does not hold:
``` 
method test()
{
    label l0
    refute false
    goto l0
}
```

**Comment by marcoeilers** [comment]
The refute plugin generates this code, which looks fine to me but makes Silicon crash:

```
method test()
{
  {
    label l0
    {
      var __plugin_refute_nondet1: Bool
      if (__plugin_refute_nondet1) {
        assert false
        inhale false
      }
    }
    goto l0
  }
}
```

**Comment by marcoeilers** [comment]
Fixed in #774.
